examples/Examples/List.hs

```diff
@@ -55,14 +55,15 @@ tests = do
           (pzipWith' (+) # integerList [1 .. 10] # integerList [1 .. 10])
             #== integerList (fmap (* 2) [1 .. 10])
     , testCase "pfoldl" $ do
+        let m :: Term _ (PInteger :--> PInteger :--> PInteger) = plam $ \x y -> x - y
         expect $
-          (pfoldl # plam (-) # 0 # integerList [1 .. 10])
+          (pfoldl # m # 0 # integerList [1 .. 10])
```

Comment:
What's the reason for this change?

Reply:
Making them coherent breaks this. In general the curried syntax just doesn't work well.

Reply:
The type of `m` also needs to be written explicitly unfortunately.